### PR TITLE
fix(cliproxy): parse Codex additional_rate_limits for Spark quota

### PR DESCRIPTION
**Pull request by yousiki**
## Summary

- The Codex `wham/usage` endpoint moved GPT-5.3 Codex Spark quota out of `code_review_rate_limit` (now always `null`) into a new `additional_rate_limits[]` array. CCS silently dropped the field, so `ccs cliproxy quota` did not show Spark windows for ChatGPT Pro accounts. Plus accounts (where the field is `null`) showed nothing extra and remain unaffected.
- Parse the new field on both CLI and Dashboard sides. Add explicit `category` / `cadence` / `featureLabel` metadata on `CodexQuotaWindow` so display can route by window type instead of label-string sniffing.
- Live Pro response shape:
```json
{
  "code_review_rate_limit": null,
  "additional_rate_limits": [{
    "limit_name": "GPT-5.3-Codex-Spark",
    "metered_feature": "codex_bengalfox",
    "rate_limit": { "primary_window": {...}, "secondary_window": {...} }
  }]
}
```

## Changes

**Server / CLI**
- `src/cliproxy/quota-types.ts`: extend `CodexQuotaWindow` with optional `category` (`'usage' | 'code-review' | 'additional'`), `cadence` (`'5h' | 'weekly'`), `featureLabel`.
- `src/cliproxy/quota-fetcher-codex.ts`: parse `additional_rate_limits[]` (snake + camel parity, `null` guarded). Each window is tagged with metadata. `buildCodexCoreUsageSummary` prefers category-based selection; legacy label-sniffing remains as fallback for cached windows.
- `src/commands/cliproxy/quota-subcommand.ts`: `getCodexCoreUsageWindows` and `getCodexWindowDisplayLabel` prefer metadata. New `prettifyCodexFeatureLabel` strips `^GPT-[\d.]+-Codex-` so `GPT-5.3-Codex-Spark` renders as `Codex Spark`.
- `tests/unit/cliproxy/quota-fetcher-codex.test.ts`: +7 tests including the real Pro response shape, the `null` guard, camelCase parity, and core-summary exclusions.

**Dashboard**
- `ui/src/lib/api-client.ts`: mirror the metadata extension on the UI type.
- `ui/src/lib/utils.ts`: `CodexWindowKind` gains `additional-5h` / `additional-weekly`; `CodexQuotaBreakdown` returns `additionalWindows`. `additional` windows are excluded from the 5h/weekly fallback pool so a Spark window with a short reset cannot displace core usage on a fresh Pro account.
- `ui/src/components/shared/quota-tooltip-content.tsx`: render Spark windows inline (between code-review and unknown).
- `ui/src/lib/i18n.ts`: `codexAdditional5h` / `codexAdditionalWeekly` / `codexAdditional` for en, zh-CN, vi, ja, using the existing `{{name}}` placeholder.
- `ui/tests/unit/ui/lib/codex-quota-breakdown.test.ts`: 18 tests covering metadata path, legacy cache path, prettify edge cases, and interpolated display labels.

## Test plan

- [x] `bun run validate` (root) — green: 2552 pass, 1 skip, 0 fail.
- [x] `cd ui && bun run validate` — green for added tests; 2 pre-existing failures in `account-visual-groups.test.ts` reproduce on `dev` (unrelated).
- [x] Live verification on a Pro account: Spark windows render with prettified labels; the `Reset schedule:` summary line still reflects only core 5h + weekly.
- [x] Live verification on a Plus account (`additional_rate_limits: null`): no Spark output, no regressions.
- [ ] Maintainer dashboard smoke test on a Pro account to confirm tooltip layout.

## Notes

- Legacy cached Spark windows lacking metadata would still hit label-sniffing classification, but this set is empty in practice: Spark only enters cache via the new builder, which always sets metadata. Pre-existing limitation, not a regression.
- Optional follow-up: have `getCodexWindowKind` require `cadence` when `category` is set (currently defaults to `5h`); harmless because the builder always emits both.

### CODE REVIEW

File: ui/src/lib/utils.ts [hunk 1]
```diff
@@ -332,12 +332,33 @@ export type CodexWindowKind =
...
+    if (w.category === 'additional') {
+      return w.cadence === 'weekly' ? 'additional-weekly' : 'additional-5h';
+    }
+    if (w.category === 'code-review') {
+      return w.cadence === 'weekly' ? 'code-review-weekly' : 'code-review-5h';
+    }
+    if (w.category === 'usage') {
+      return w.cadence === 'weekly' ? 'usage-weekly' : 'usage-5h';
+    }
+    // No category metadata -> fall through to label sniffing.
```
[the remark this answers is not in the document]
Comment (yousiki) [reply ui/src/lib/utils.ts]: Done in a90aadd2 — `getCodexWindowKind` now falls through to label sniffing when `category` is set without `cadence`.

File: src/commands/cliproxy/quota-subcommand.ts [hunk 2]
```diff
@@ -272,12 +275,46 @@ function inferCodeReviewCadence(
...
+  if (window.category === 'usage') {
+    if (window.cadence === '5h') return '5h usage limit';
+    if (window.cadence === 'weekly') return 'Weekly usage limit';
+  }
+
+  if (window.category === 'additional') {
+    const pretty = prettifyCodexFeatureLabel(window.featureLabel || 'Additional');
+    if (window.cadence === '5h') return `${pretty} (5h)`;
+    if (window.cadence === 'weekly') return `${pretty} (weekly)`;
+    return pretty;
```
[the remark this answers is not in the document]
Comment (yousiki) [reply src/commands/cliproxy/quota-subcommand.ts]: Done in a90aadd2 — fallback chain is now `window.featureLabel || window.label || 'Additional'`, so partial cached windows preserve their raw upstream label.

**Comment by yousiki** [comment]
Pushed 0e60293 to fix the build failure. Verified locally with `bun run validate:ci-parity`.

**Comment by kaitranntt** [comment]
@yousiki thanks for the strong follow-through here. Great job on the fix and the careful iterations.
